build.gradle.kts

```diff
@@ -0,0 +1,95 @@
+plugins {
+	id("fabric-loom") version "0.7-SNAPSHOT"
+	`maven-publish`
+}
+
+val sourceCompatibility = JavaVersion.VERSION_1_8
+val targetCompatibility = JavaVersion.VERSION_1_8
+
+val archives_base_name: String by project
+val mod_version: String by project
+val maven_group: String by project
+
+
+repositories {
+	// Add repositories to retrieve artifacts from in here.
+	// You should only use this when depending on other mods because
+	// Loom adds the essential maven repositories to download Minecraft and libraries from automatically.
+	// See https://docs.gradle.org/current/userguide/declaring_repositories.html
+	// for more information about repositories.
+}
+
+val minecraft_version: String by project
+val yarn_mappings: String by project
+val loader_version: String by project
+
+val fabric_version: String by project
+
+dependencies {
+	// To change the versions see the gradle.properties file
+	minecraft("com.mojang:minecraft:${minecraft_version}")
+	mappings("net.fabricmc:yarn:${yarn_mappings}:v2")
+	modImplementation("net.fabricmc:fabric-loader:${loader_version}")
+
+	// Fabric API. This is technically optional, but you probably want it anyway.
+	modImplementation("net.fabricmc.fabric-api:fabric-api:${fabric_version}")
+
+	// PSA: Some older mods, compiled on Loom 0.2.1, might have outdated Maven POMs.
+	// You may need to force-disable transitiveness on them.
+}
+
+tasks {
+	withType<JavaCompile> {
+		// ensure that the encoding is set to UTF-8, no matter what the system default is
+		// this fixes some edge cases with special characters not displaying correctly
+		// see http://yodaconditions.net/blog/fix-for-java-file-encoding-problems-with-gradle.html
+		// If Javadoc is generated, this must be specified in that task too.
```

Comment:
Should just link to https://docs.gradle.org/current/dsl/org.gradle.api.tasks.compile.CompileOptions.html than some random blog

Reply:
this is same for the original

Reply:
I guess he just copied the original but I agree the wall of text could be shortened down

Reply:
But if you do carry out a big change like switching to kotlin, it's better to fix the overdue technical debt at the same time.